src/main/js/field/TableDisplay.js

```diff
@@ -297,33 +272,58 @@ class TableDisplay extends Component {
         this.props.clearModalFieldState();
         this.props.editState(currentRowSelected);
         this.setState({
-            currentRowSelected,
-            modificationState: MODIFICATION_STATE.EDIT
+            currentRowSelected
         });
     }
 
     editButtonClick(cell, row) {
-        return <IconTableCellFormatter handleButtonClicked={this.editButtonClicked} currentRowSelected={row} buttonIconName="pencil-alt" buttonText="Edit" />;
+        return (<IconTableCellFormatter
+            handleButtonClicked={this.editButtonClicked}
+            currentRowSelected={row}
+            buttonIconName="pencil-alt"
+            buttonText="Edit"
+        />);
     }
 
     copyButtonClicked(currentRowSelected) {
+        currentRowSelected.id = null;
         this.props.editState(currentRowSelected);
         this.setState({
-            currentRowSelected,
-            modificationState: MODIFICATION_STATE.COPY
+            currentRowSelected
         });
     }
 
     copyButtonClick(cell, row) {
-        return <IconTableCellFormatter handleButtonClicked={this.copyButtonClicked} currentRowSelected={row} buttonIconName="copy" buttonText="Copy" />;
+        return (<IconTableCellFormatter
+            handleButtonClicked={this.copyButtonClicked}
+            currentRowSelected={row}
+            buttonIconName="copy"
+            buttonText="Copy"
+        />);
     }
 
     render() {
         const tableColumns = this.createTableColumns();
-        tableColumns.push(<TableHeaderColumn dataField="" width="48" columnClassName="tableCell" dataFormat={this.editButtonClick} thStyle={{ "text-align": "center" }}>Edit</TableHeaderColumn>);
-        tableColumns.push(<TableHeaderColumn dataField="" width="48" columnClassName="tableCell" dataFormat={this.copyButtonClick} thStyle={{ "text-align": "center" }}>Copy</TableHeaderColumn>);
-
-        const { selectRowBox, sortName, sortOrder, autoRefresh, tableMessage, newButton, deleteButton, data, tableSearchable } = this.props;
+        tableColumns.push(<TableHeaderColumn
+            dataField=""
+            width="48"
+            columnClassName="tableCell"
+            dataFormat={this.editButtonClick}
+            thStyle={{ textAlign: 'center' }}
+        >Edit
```

Comment:
The formatting here is a little strange to me. Edit on its own line would be a little clearer